satellite/console/service.go

```diff
@@ -175,16 +174,18 @@ func (s *Service) CreateUser(ctx context.Context, user CreateUser, tokenSecret R
 		}
 
 		if currentReward != nil {
-			// User can only earn credits after activating their account. Therefore, we set the credits to 0 on registration
-			newCredit := UserCredit{
-				UserID:        u.ID,
-				OfferID:       currentReward.ID,
-				ReferredBy:    nil,
-				CreditsEarned: currency.Cents(0),
-				ExpiresAt:     time.Now().UTC().AddDate(0, 0, currentReward.InviteeCreditDurationDays),
+			var refID *uuid.UUID
+			if refUserID != "" {
+				refID, err = uuid.Parse(refUserID)
+				if err != nil {
+					return errs.New(internalErrMsg)
```

Comment:
you don't need to change but typically I prefer to just return `http.StatusInternalServerError`